build/release.mk

```diff
@@ -190,6 +190,17 @@ else
 	done
 endif
 
+	@# Products
+
+	@if [ -d $(BUILD_BOARDS_DIR) ] ; then \
+		echo "Copying web app files for Boards product"; \
+		mkdir -p $(DIST_PATH_GENERIC)/client/products/boards; \
+		cp -R $(BUILD_BOARDS_DIR)/mattermost-plugin/webapp/dist/* $(DIST_PATH_GENERIC)/client/products/boards/; \
+	else \
+		echo "Unable to find files for Boards product. Please ensure that the Focalboard repository is checked out alongside the server and run 'make build-product' in it."; \
```

Comment:
Is it intended that the build fails now if the boards files aren't around? I just want to build and package the mattermost server without the boards stuff as before.

Reply:
Yeah, that was intended, but we could always consider changing it. We're in the process of integrating Boards more into Mattermost instead of having it run as a plugin, and while that's behind a feature flag when running MM in 7.5, it'll eventually become the only way to run Boards in the next few months or so, so it would at least be recommended to have at this point in the packaging scripts. The main reason it errors out is so that we wouldn't accidentally forget those files when packaging our builds, and I did it here since I figured we were the only one using this particular Makefile. I take it you're packaging a custom build using these as well though?

That said, we've also been hoping on not requiring those for development builds at least, so we could possibly make that work for custom production builds as well.

I'll bring this up for discussion next week to see how other people on the team would feel about doing that since  we haven't firmly committed yet to whether or not things like Boards will be a required part of everything in the future since they are likely going to become more integrated with the rest of Mattermost over time.

Reply:
I understand about the concerns of bloat. The idea of it being mandatory has been thrown around once or twice, but it's definitely not something we've committed to. There's people on the team who wouldn't want that to happen either, so you're not alone in that.

As for patching that, currently, there'd be no repercussions for doing that, but in a release or two when we turn the feature on by default (and then on fully) in production builds, the web app will start printing error messages to the console since it'll be trying to load the missing files.

If you're also also building the web app from source with that, maybe we can add an environment variable for both these that'll disable trying to compile the Boards files for the web app? That'd save you from having to patch anything, and it'd still include everything by default like we want on our end. Plus, since we'd have code referencing that on our end, it'd be much less likely that we accidentally remove it without having a good reason to.

Reply:
> If you're also also building the web app from source with that, maybe we can add an environment variable for both these that'll disable trying to compile the Boards files for the web app?

Yep the ebuild builds the webapp and the server. A ENV switch would be ideal as Gentoo's package manager has a feature called `use-flags` which are basically feature flags. Could easily add a use-flag that pulls in the boards functionality for people who want it.

Reply:
+1 no keeping Boards (or Playbooks) optional. I'm using Mattermost purely as a communication tool and want to avoid feature creep.